### PR TITLE
Refactor: replace the bash code with ruby

### PR DESCRIPTION
**Pull request by lilirui**
@tw4dl @zquestz 
FYI, as we discussed in Friday, I replace all the bash scripts with ruby code. The good thing is right now we support both vmware and virtualbox without changing vagrant file.
vagrant up --provider virtualbox
vagrant up --provider vmware_fusion or vmware_workstation

### CODE REVIEW

**Comment by lilirui** [comment]
Any feedbacks?

**Comment by tw4dl** [comment]
Tried and LGTM.

**Comment by lilirui** [comment]
@zquestz  does this LGTY?

File: vagrant/Vagrantfile [hunk 1]
```diff
@@ -1,14 +1,29 @@
...
+    end
+
+    default.vm.provider :virtualbox do |v|
+      helper.virtualbox_options(v)
+    end
+
+    helper.share_folder(default)
+    helper.scripts(default)
+    end
+end
```
Comment (zquestz) [inline vagrant/Vagrantfile]: Add a newline, this can be added in Sublime Text =)

File: vagrant/setup.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                "tee /var/log/continuum-install.log"
+    err, out = exec_cmd(setup_cmd)
+    on_error(err) if err != ""
+    puts out
+  end
+
+  def update_config
+    err, out = exec_cmd('cat /CONTINUUM_DNS_TOKEN') 
+    on_error(err) if err != ""
+    @config["token"] = out.delete!("\n")
```
Comment (zquestz) [inline vagrant/setup.rb]: You can just use ruby's strip function.

File: vagrant/setup.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    cmd = "cat /var/log/continuum-install.log | grep 'apc target' | awk '{print $3}' | cut -d'.' -f 1"
+    err, out = exec_cmd(cmd)
+    on_error(err) if err != ""
+    @config["name"] = out.delete!("\n")
```
Comment (zquestz) [inline vagrant/setup.rb]: ditto.

File: vagrant/vagrant_helper.rb [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+  def workstation?
+    @provider_type == 'vmware_workstation'
+  end
+
+  def virtualbox?
+    @provider_type == 'virtualbox'
+  end
+
+end
```
Comment (zquestz) [inline vagrant/vagrant_helper.rb]: newline =)

**Comment by zquestz** [comment]
Tiny adjustments, but LGTM.

**Comment by lilirui** [comment]
All the feedbacks are addressed.

**Comment by zquestz** [comment]
LGTM

**Pull request merged**